tests/test_encoding.py

```diff
@@ -287,7 +287,10 @@ def test_encode_overlapping(self) -> None:
             parameters=NamedItemList([param1, param2, param3]),
             byte_size=None,
         )
-        self.assertEqual(req.encode(), bytearray([0x12, 0x34, 0x56]))
+
+        with self.assertRaises(EncodeError):
```

Comment:
This should not throw, only warn

Reply:
it only raises an exception when in strict mode...